### PR TITLE
fix #9483

### LINKED ISSUE #9483: oom messes up the maps
screen shot 2018-02-23 at 18 25 13

### PR DESCRIPTION
**Pull request by condret**

### CODE REVIEW

File: libr/core/cmd_open.c [hunk 1]
```diff
@@ -782,35 +778,56 @@ static void cmd_open_map(RCore *core, const char *input) {
...
+		return true;
+	}
+
+	if (strstr (desc->uri, "://")) {
+		return true;
+	}
+
+	size = r_io_desc_size (desc);
+
+	uri = r_str_newf("malloc://%"PFMT64u, size);
```
[the remark this answers is not in the document]
Comment (condret) [reply libr/core/cmd_open.c]: i thought putting it to the top makes code more readable

File: libr/core/cmd_open.c [hunk 2]
```diff
@@ -782,35 +778,56 @@ static void cmd_open_map(RCore *core, const char *input) {
...
+// if malloc fails, we can just abort the loop by returning false
+	if (!buf) {
+		free (uri);
+		return false;
+	}
+
+	ndesc = r_io_open_nomap (io, uri, R_IO_RW, 0);
+	free (uri);
+	if (!ndesc) {
 		free (buf);
```
Comment (radare) [inline libr/core/cmd_open.c]: memory leak uri not freed

File: libr/core/cmd_open.c [hunk 3]
```diff
@@ -782,35 +778,56 @@ static void cmd_open_map(RCore *core, const char *input) {
...
+
+	if (r_io_desc_is_blockdevice (desc) || r_io_desc_is_dbg (desc)) {
+		return true;
+	}
+
+	if (strstr (desc->uri, "://")) {
+		return true;
+	}
+
+	size = r_io_desc_size (desc);
```
Comment (radare) [inline libr/core/cmd_open.c]: define and assign in the same line

File: libr/core/cmd_open.c [hunk 4]
```diff
@@ -782,35 +778,56 @@ static void cmd_open_map(RCore *core, const char *input) {
...
-		break;
+		return false;
+	}
+
+	r_io_desc_read_at (desc, 0LL, buf, (int)size);	//that cast o_O
+	r_io_desc_write_at (ndesc, 0LL, buf, (int)size);
+	free (buf);
+	r_io_desc_exchange (io, desc->fd, ndesc->fd);
+
+	r_io_desc_close (desc);
```
Comment (radare) [inline libr/core/cmd_open.c]: free uri too

Comment (radare) [reply libr/core/cmd_open.c]: nvm its freed some lines above. mabe use R_FREE ?

Comment (condret) [reply libr/core/cmd_open.c]: using R_FREE would be pointless here, since it is never used again

File: libr/core/cmd_open.c (hunk 2 above)
Comment (condret) [reply libr/core/cmd_open.c]: prove it

File: libr/core/cmd_open.c [hunk 5]
```diff
@@ -782,35 +778,56 @@ static void cmd_open_map(RCore *core, const char *input) {
...
+
+	size = r_io_desc_size (desc);
+
+	uri = r_str_newf ("malloc://%"PFMT64u, size);
+	if (!uri) {
+		return false;
+	}
+
+	buf = malloc (size);
+// if malloc fails, we can just abort the loop by returning false
```
Comment (MaskRay) [inline libr/core/cmd_open.c]: The code self explains, no need for a comment.

Comment (condret) [reply libr/core/cmd_open.c]: only for advanced programmers, like you

**Comment by condret** [comment]
why does this fail travis?

**Comment by radare** [comment]
please rebase now that i "fixed" travis for you

**Pull request merged**